test/OpenTelemetry.AutoInstrumentation.Tests/Loading/LazyInstrumentationLoaderTests.cs

```diff
@@ -0,0 +1,140 @@
+// <copyright file="LazyInstrumentationLoaderTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using OpenTelemetry.AutoInstrumentation.Configuration;
+using OpenTelemetry.AutoInstrumentation.Loading;
+using Xunit;
+
+namespace OpenTelemetry.AutoInstrumentation.Tests.Loading
+{
+    public class LazyInstrumentationLoaderTests
+    {
+        [Fact]
+        public void LazyInstrumentationLoader_DefaultLoad()
+        {
+            var detectors = new Dictionary<TracerInstrumentation, Func<AssemblyLoadDetector>>()
+            {
+                { default(TracerInstrumentation), () => new DummyDetector() }
+            };
+
+            var loader = CreateLoader(new NameValueCollection(), detectors);
+
+            CreateDummyAssembly();
+
+            var instrumentation = loader.Instrumentations.FirstOrDefault();
+
+            Assert.True(instrumentation is DummyInstrumentation);
+        }
+
+        [Fact]
+        public void LazyInstrumentationLoader_InstrumentationNotEnabled()
+        {
+            var detectors = new Dictionary<TracerInstrumentation, Func<AssemblyLoadDetector>>()
+            {
+                { TracerInstrumentation.HttpClient, () => new DummyDetector() }
+            };
+
+            var config = new NameValueCollection()
+            {
+                { ConfigurationKeys.Traces.Instrumentations, nameof(TracerInstrumentation.AspNet) }
```

Comment:
can it be e.g. `OtherLibrary` instead of  `nameof(TracerInstrumentation.AspNet)`

Reply:
That would require turning away from enum based values. Perhaps I'll add some comments to the code?